includes/admin/class-wc-admin-menus.php

```diff
@@ -116,6 +106,34 @@ public function settings_page_init() {
 		do_action( 'woocommerce_settings_page_init' );
 	}
 
+	/**
+	 * Handle saving of settings.
+	 *
+	 * @return void
+	 */
+	public function save_settings() {
+		global $current_tab, $current_section;
+
+		// We should only save on the settings page.
+		if ( ! is_admin() || ! isset( $_GET['page'] ) || 'wc-settings' !== $_GET['page'] ) { // phpcs:ignore WordPress.Security.NonceVerification.NoNonceVerification
```

Comment:
Just wondering if it isn't better to check the nonce earlier than in `WC_Admin_Settings::save()`. Perhaps it doesn't matter much, just following the 'validate input early' rule I memorised from the Plugin developer handbook :)
On the other hand, maybe it would just introduce more problems, as public method `WC_Admin_Settings::save()` already verifies nonce, so removing it from that one could introduce problems and we don't want to perform double work.